packages/workflow/src/utils/duplicateChecker.ts

```diff
@@ -99,44 +106,78 @@ async function hasBirthDuplicates(
     motherDoB: birthRegDetails.mother?.birthDate
   })
 
-  return !res || res.length > 0
+  return res
 }
 
-async function hasDeathDuplicates(
+async function findDeathDuplicateIds(
   authHeader: IAuthHeader,
   deathRegDetails: DeathRegistration
-) {
+): Promise<{ id: string; trackingId: string }[]> {
   if (!deathRegDetails || !deathRegDetails.deceased) {
-    return false
+    return []
   }
 
-  const res = await findDeathDuplicates(authHeader, {
+  const res = await searchDeathDuplicates(authHeader, {
     deceasedFirstNames: deathRegDetails.deceased?.name?.[0]?.firstNames,
     deceasedFamilyName: deathRegDetails.deceased?.name?.[0]?.familyName,
     deceasedIdentifier: deathRegDetails.deceased?.identifier?.[0]?.id,
     deceasedDoB: deathRegDetails.deceased?.birthDate,
     deathDate: deathRegDetails.deceased?.deceased?.deathDate
   })
 
-  return !res || res.length > 0
+  return res
 }
 
-export async function hasDuplicates(
+export async function findDuplicateIds(
   registrationDetails: BirthRegistration | DeathRegistration,
   authHeader: IAuthHeader,
   event: EVENT_TYPE
 ) {
   if (event === EVENT_TYPE.BIRTH) {
-    return hasBirthDuplicates(
+    return findBirthDuplicateIds(
       authHeader,
       registrationDetails as BirthRegistration
     )
   } else if (event === EVENT_TYPE.DEATH) {
-    return hasDeathDuplicates(
+    return findDeathDuplicateIds(
       authHeader,
       registrationDetails as DeathRegistration
     )
   }
   // NOT IMPLEMENTED FOR MARRIAGE
-  return false
+  return []
+}
+
+export function updateTaskWithDuplicateIds(
+  task: SavedTask,
+  duplicateIds: { id: string; trackingId: string }[]
+) {
+  const extension: Extension = {
+    url: FLAGGED_AS_POTENTIAL_DUPLICATE,
+    valueString: duplicateIds
+      .map((duplicate) => duplicate.trackingId)
+      .toString()
+  }
+  return addExtensionsToTask(
+    {
+      ...task,
+      lastModified: new Date().toISOString()
+    },
+    [extension]
+  ) as SavedTask
+}
+
+export function updateCompositionWithDuplicateIds(
+  composition: Composition,
+  duplicateIds: { id: string; trackingId: string }[]
+) {
+  const relatesTo: NonNullable<Composition['relatesTo']> = duplicateIds.map(
+    (duplicate) => ({
+      code: 'duplicate',
+      targetReference: {
+        reference: `Composition/${duplicate.id}`
+      }
+    })
+  )
+  return addRelatesToToComposition(composition, relatesTo) as SavedComposition
```

Comment:
`as SavedComposition` doesn't look right as you are getting a `Composition` as argument